### PR TITLE
[PUPIL-864] Fix accessibility issue on OakQuizHint

### PR DESCRIPTION
**Pull request by BTitterington**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

There was an issue with the `aria-describedby` referring to an id of an element that is only rendered when the button is clicked.

The fix is to place an additional hidden div with that id next to the component that contains the hint.

In addition, I noticed that with the `id` hardcoded it would mean we could only use the OakQuizHint once on a page. I adjusted it so the id is passed in. This also meant OakLessonBottomNav needed to include an `id` on the usage of OakQuizHint.

### CODE REVIEW

File: src/components/organisms/pupil/OakQuizHint/OakQuizHint.tsx [hunk 1]
```diff
@@ -2,35 +2,37 @@ import React, { MouseEventHandler, ReactNode, useState } from "react";
...
    */
   hint: ReactNode;
+  id: string;
 };
 
 /**
  * Presents a button which will show a hint when clicked
  */
-export const OakQuizHint = ({ hint }: OakQuizHintProps) => {
+export const OakQuizHint = ({ hint, id }: OakQuizHintProps) => {
```
Comment (kimon-satan) [inline src/components/organisms/pupil/OakQuizHint/OakQuizHint.tsx:18]: might be worth defaulting the id in any case ?

File: src/components/organisms/pupil/OakLessonBottomNav/OakLessonBottomNav.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ export const OakLessonBottomNav = ({
       );
       break;
     case !!hint:
-      content = <OakQuizHint hint={hint} />;
+      content = <OakQuizHint hint={hint} id="quiz-hint" />;
```
Comment (kimon-satan) [inline src/components/organisms/pupil/OakLessonBottomNav/OakLessonBottomNav.tsx:44]: A default would make this change unecessary

**Review by kimon-satan — APPROVED** [review]

File: src/components/organisms/pupil/OakQuizHint/OakQuizHint.tsx (hunk 1 above)
Comment (BTitterington) [reply src/components/organisms/pupil/OakQuizHint/OakQuizHint.tsx:18]: I considered that, but then I'd have to make `id` optional and figured it is better to draw attention to the fact an `id` it is needed (if it ever gets used elsewhere). The other option was to use a generated guid for the id, but I figured I was overthinking it 🙂

**Review by RBrownsett — APPROVED** [review]
Approved - now passing accessibility  checker 
image

**Pull request merged**